addons/product/models/product_product.py

```diff
@@ -96,9 +96,9 @@ def _compute_can_image_variant_1024_be_zoomed(self):
     def _set_template_field(self, template_field, variant_field):
         for record in self:
             if (
-                # We are trying to remove a field from the variant even though it is already
-                # not set on the variant, remove it from the template instead.
-                (not record[template_field] and not record[variant_field])
+                # We are trying to remove a field. Remove it from the template
+                # and from the variant.
+                not record[template_field]
```

Comment:
@loco-odoo I still don't understand this change, I just tested, this just does not make sense :grimacing: Have a product with 10 variants, configure an image for the template (base case), then set up the image for like 3 variants. Then, enter edit mode remove the image you set up for one of those 3 variants => bam now 8 variants do not have an image anymore (the current one and the 7 others that had no specific image)... I really don't think this is what we want, we don't want the template image to be removed when editing a variant, ever, or I am missing something... :shrug:

There probably is still a spec problem about all this: if we keep the current 16.0 behavior, removing a variant image would revert it back to the template image (seems good) but... what should hitting the "remove" button again do? I don't know. Can you check with PAJU what we want for all use cases? :grimacing: 

- Do we show the "remove" button always? (last time she said yes but I don't think all usecases were considered (like the one I described above)).
- What does it to in all the possible use case:
    - Remove a variant's specific image
    - Remove a variant's image which is in fact the template image (while it is used by other variants or not)
    - Remove the template image (while it is used by many variants or not)
    - More ?

Reply:
@paju-odoo, can you confirm that this is what you want for the remove of the main image :

-For a product with no variants, the image of the product falls back on the image of the template so removing the image of the product leads to removing the image of the template (see video 1).

-For a product with multiples variants :

1. If there is no specific image for the variant : the product image of the variant falls back on the template image. Removing a product image variant leads to removing the template image → the template and all the other variants with no specific image loose their image (see video 2).
2. If there is a specific image for the variant, removing the product variant image leads to fallback on the template image. (see video 3). Note than clicking again on the "Remove button" leads to the same behavior than the one just above.

Thanks in advance!


[Screencast from 06-20-2023 05:25:36 PM.webm](https://github.com/odoo/odoo/assets/114914971/210b85b3-753b-44b1-9930-1745404f6d64)


[Screencast from 06-20-2023 05:27:33 PM.webm](https://github.com/odoo/odoo/assets/114914971/8024bd89-8d10-40f7-8d60-0dfd95caba9d)


[Screencast from 06-20-2023 05:31:04 PM.webm](https://github.com/odoo/odoo/assets/114914971/d7ac66b2-42b5-44c8-b621-a58da5915d44)

Reply:
> 
@loco-odoo 

Indeed, that's the expected behavior and how it's been working so far. If no specific image has been set on a variant, the template's image is in fact the variant's image, so removing it will remove the image from the template and all other variants using it. 

I'm not saying this is the best behavior, but it is the current one and we should not try to change it in regards with this fix. 

We'll have a look into it in the eCommerce and try to come up with a better solution in the future, but please keep it as it is :)  

cc @qsm-odoo

Reply:
@loco-odoo I am bit lost: if this is the current behavior, why do we need your diff here in `_set_template_field` to make it work? ^^" Or is it just about the "dirty" flag issue and then your diff would not be needed?

Reply:
> Or is it just about the "dirty" flag issue and then your diff would not be needed?

Exactly, after the "dirty" flag issue fix, the diff in `_set_template_field` will not be needed anymore :+1:

Reply:
Hi @loco-odoo,

Sorry for my late reply, I wasn't working yesterday... I'd prefer the latter option, have the template image displayed in the last position :)